flow/SystemMonitor.cpp

```diff
@@ -301,6 +301,7 @@ SystemStatistics customSystemMonitor(std::string const& eventName, StatisticsSta
 			    .detail("DCID", machineState.dcId)
 			    .detail("ZoneID", machineState.zoneId)
 			    .detail("MachineID", machineState.machineId)
+			    .detail("DatahallID", machineState.datahallId)
```

Comment:
It feels at this point we should add `toString` to `SystemMonitorMachineState` and print all required info?

Reply:
IMO this may be useful to add in a separate PR on `main`. But there's also value in having separate fields in the trace event for each component of the locality.

Reply:
Agreed. The point I was driving is we are adding all fields, so maybe adding `toString()` might be helpful for both code quality as well as at client usage site where we print all relevant values as a single unit. 

But, for sure this can be done as separate PR.